### PR TITLE
feat(widgets): add name style settings for line/table + edit label in config panel

### PR DESCRIPTION
**Pull request by ssjagad**
## Overview
- Add name property to style settings object for non-line widgets
- Add name property style chart options for line chart
- created label editing component for config panel
- users can now edit names of individual properties from config panel or revert to original name

https://github.com/awslabs/iot-app-kit/assets/145582655/e5653326-5aed-4d6b-840e-6998ffc0e7e6

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/components/chart/chartOptions/useChartConfiguration.ts [hunk 1]
```diff
@@ -89,9 +90,15 @@ const toDataStreamMetaData = (
         ({ id: seriesId }) => seriesId === id
       ) ?? { appKitColor: color };
       const colorUsed = (foundSeries as GenericSeries).appKitColor;
+
+      let styledName;
+      if (refId && styleSettings) {
+        styledName = styleSettings[refId]?.name;
```
Comment (jmbuss) [inline packages/react-components/src/components/chart/chartOptions/useChartConfiguration.ts:96]: We can probably turn this into a ternary and get rid of some code!

File: packages/react-components/src/components/chart/baseChart.tsx [hunk 2]
```diff
@@ -98,7 +98,7 @@ const BaseChart = ({
     thresholds,
     utilizedViewport,
     visibleData,
-  } = useVisualizedDataStreams(queries, viewport);
+  } = useVisualizedDataStreams(queries, viewport, undefined);
```
Comment (jmbuss) [inline packages/react-components/src/components/chart/baseChart.tsx:101]: is it important to pass undefined here? Are we using this hook in any other components? Maybe we can just remove that param entirely?

**Review by jmbuss — COMMENTED** [review]
Looks good to me! I think there are a few things commented out that you can just delete, and then I had a few questions.

File: packages/react-components/src/components/chart/baseChart.tsx (hunk 2 above)
Comment (ssjagad) [reply packages/react-components/src/components/chart/baseChart.tsx]: Nope, that should be removed since im not passing styleSettings as a parameter there

**Review by corteggiano — APPROVED** [review]

File: packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts [hunk 3]
```diff
@@ -30,13 +31,13 @@ export const useVisualizedDataStreams = (
...
     (lastUpdatedBy ? viewport : passedInViewport || viewport) ??
     DEFAULT_VIEWPORT;
-
   const { dataStreams, thresholds: queryThresholds } = useTimeSeriesData({
     viewport: utilizedViewport,
     queries,
     settings: {
       fetchFromStartToEnd: true,
     },
+    // styles: styleSettings,
```
Comment (jmbuss) [inline packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts:40]: 1 more comment :D

File: packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ import isEqual from 'lodash.isequal';
 import { useTimeSeriesData } from '../../../hooks/useTimeSeriesData';
 import { useViewport } from '../../../hooks/useViewport';
 import { DEFAULT_VIEWPORT, StreamType } from '../../../common/constants';
+// import { ChartStyleSettings } from '../types';
```
Comment (jmbuss) [inline packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts:14]: A little cleanup here :D

**Pull request merged**